[bug] `context-name` rule shouldn't be enabled for react v18

### Describe the bug

Quoted from the [docs](https://www.eslint-react.xyz/docs/rules/naming-convention-context-name):



> Enforces the context name to be a valid component name with the suffix Context.
> 
> In React 19, you can render <Context> as a provider instead of <Context.Provider>. This rule enforces that the context has a valid component name with the suffix Context.

Therefore, for React version < 19, this rule is not necessary, and it would be nice to disable it by default (i.e., not included in `recommended`).

### Reproduction

https://github.com/issueset/eslint-react-context-name-issue.git

 

### Expected behavior

No ESLint warning shown 

### Platform and versions

```Text
$ pnpm run lint

src/theme/ThemeContext.tsx
  42:7  warning  A context name must be a valid component name with the suffix 'Context'  @eslint-react/naming-convention/context-name
```

### Stack trace

```Text
N/A
```

### Additional context

N/A

--- a/packages/plugins/eslint-plugin-react-naming-convention/package.json
+++ b/packages/plugins/eslint-plugin-react-naming-convention/package.json
@@ -53,6 +53,7 @@
     "@typescript-eslint/type-utils": "^8.51.0",
     "@typescript-eslint/types": "^8.51.0",
     "@typescript-eslint/utils": "^8.51.0",
+    "compare-versions": "^6.1.1",
     "string-ts": "^2.3.1",
     "ts-pattern": "^5.9.0"
   },

--- a/packages/plugins/eslint-plugin-react-naming-convention/src/rules/context-name.spec.ts
+++ b/packages/plugins/eslint-plugin-react-naming-convention/src/rules/context-name.spec.ts
@@ -51,6 +51,17 @@ ruleTester.run(RULE_NAME, rule, {
       `,
       errors: [{ messageId: "invalidContextName" }],
     },
+    {
+      code: tsx`
+        const themecontext = React.createContext("");
+      `,
+      errors: [{ messageId: "invalidContextName" }],
+      settings: {
+        "react-x": {
+          version: "19.0.0",
+        },
+      },
+    },
   ],
   valid: [
     ...allFunctions,
@@ -71,5 +82,15 @@ ruleTester.run(RULE_NAME, rule, {
     tsx`
       ctxs.ThemeContext = createContext("");
     `,
+    {
+      code: tsx`
+        const themecontext = React.createContext("");
+      `,
+      settings: {
+        "react-x": {
+          version: "18.2.0",
+        },
+      },
+    },
   ],
 });

--- a/packages/plugins/eslint-plugin-react-naming-convention/src/rules/context-name.ts
+++ b/packages/plugins/eslint-plugin-react-naming-convention/src/rules/context-name.ts
@@ -1,7 +1,8 @@
 import { getInstanceId, isComponentName, isCreateContextCall } from "@eslint-react/core";
-import type { RuleContext, RuleFeature } from "@eslint-react/shared";
+import { type RuleContext, type RuleFeature, getSettingsFromContext } from "@eslint-react/shared";
 import { AST_NODE_TYPES as T } from "@typescript-eslint/types";
 import type { RuleListener } from "@typescript-eslint/utils/ts-eslint";
+import { compare } from "compare-versions";
 import { P, match } from "ts-pattern";
 
 import { createRule } from "../utils";
@@ -31,6 +32,9 @@ export default createRule<[], MessageID>({
 export function create(context: RuleContext<MessageID, []>): RuleListener {
   // Fast path: skip if `createContext` is not present in the file
   if (!context.sourceCode.text.includes("createContext")) return {};
+  const { version } = getSettingsFromContext(context);
+  // Skip if React version is less than 19.0.0
+  if (compare(version, "19.0.0", "<")) return {};
   return {
     CallExpression(node) {
       if (!isCreateContextCall(context, node)) return;

--- a/packages/plugins/eslint-plugin-react-x/src/rules/jsx-key-before-spread.mdx
+++ b/packages/plugins/eslint-plugin-react-x/src/rules/jsx-key-before-spread.mdx
@@ -16,7 +16,6 @@ react-x/jsx-key-before-spread
 
 **Presets**
 
-`x`
 `recommended`
 `recommended-typescript`
 `recommended-type-checked`

--- a/packages/plugins/eslint-plugin-react-x/src/rules/no-forward-ref.ts
+++ b/packages/plugins/eslint-plugin-react-x/src/rules/no-forward-ref.ts
@@ -41,8 +41,8 @@ export function create(context: RuleContext<MessageID, []>): RuleListener {
   if (!context.sourceCode.text.includes("forwardRef")) {
     return {};
   }
-  // Skip if React version is less than 19.0.0
   const { version } = getSettingsFromContext(context);
+  // Skip if React version is less than 19.0.0
   if (compare(version, "19.0.0", "<")) {
     return {};
   }

--- a/packages/plugins/eslint-plugin-react-x/src/rules/no-use-context.ts
+++ b/packages/plugins/eslint-plugin-react-x/src/rules/no-use-context.ts
@@ -39,6 +39,7 @@ export function create(context: RuleContext<MessageID, []>): RuleListener {
   // Fast path: skip if `useContext` is not present in the file
   if (!context.sourceCode.text.includes("useContext")) return {};
   const settings = getSettingsFromContext(context);
+  // Skip if React version is less than 19.0.0
   if (compare(settings.version, "19.0.0", "<")) {
     return {};
   }

--- a/packages/plugins/eslint-plugin/README.md
+++ b/packages/plugins/eslint-plugin/README.md
@@ -191,8 +191,8 @@ This project is, and will remain, 90% of its code written by humans.
 
 Contributions are welcome!
 
-Please follow our [contributing guidelines](https://github.com/Rel1cx/eslint-react/tree/main/.github/CONTRIBUTING.md).
+Please follow our [contributing guidelines](https://github.com/Rel1cx/eslint-react/tree/react-version/.github/CONTRIBUTING.md).
 
 ## License
 
-This project is licensed under the MIT License - see the [LICENSE](https://github.com/Rel1cx/eslint-react/tree/main/LICENSE) file for details.
+This project is licensed under the MIT License - see the [LICENSE](https://github.com/Rel1cx/eslint-react/tree/react-version/LICENSE) file for details.

